.golangci.yml

```diff
@@ -1,79 +1,102 @@
+version: "2"
 run:
-  timeout: 5m
   tests: false
-  
 linters:
-  disable-all: true
+  default: none
   enable:
-    - errcheck
     - copyloopvar
-    - gci
-    - gofmt
-    - gosimple
+    - errcheck
     - govet
     - ineffassign
     - misspell
     - nilerr
     - nolintlint
     - revive
     - staticcheck
-    - typecheck
     - unused
-
-linters-settings:
-  gci:
-    sections:
-      - standard
-      - default
-      - prefix(github.com/algorand)
-      - prefix(github.com/algorand/go-algorand-sdk)
-    section-separators:
-      - newLine
-  nolintlint:
-    # require naming a specific linter X using //nolint:X
-    require-specific: true
-    # require comments like "//nolint:errcheck // Explanation of why we are ignoring linter here..."
-    require-explanation: true
-
-severity:
-  default-severity: error
-
+  settings:
+    nolintlint:
+      # require comments like "//nolint:errcheck // Explanation of why we are ignoring linter here..."
+      require-explanation: true
+      # require naming a specific linter X using //nolint:X
+      require-specific: true
+  exclusions:
+    generated: lax
+    rules:
+      # revive: types is an acceptable package name in this codebase
+      - linters:
+          - revive
+        path: types/
+        text: 'var-naming: avoid meaningless package names'
+      # Examples directory is allowed to skip error checking for brevity
+      - linters:
+          - errcheck
+        path: examples/
+      # Examples directory does not require comments on exported functions
+      - linters:
+          - revive
+        path: examples/
+        text: 'exported: exported function .* should have comment or be unexported'
+      # Test utilities and helpers may have code that look unused, but is being used in another file
+      - path: test/helpers.go
+        text: is unused
+      - path: test/utilities.go
+        text: is unused
+      # Ignore unused fields in types that are copied from go-algorand
+      - path: types/
+        text: is unused
+      # ignore issues about the way we use _struct fields to define encoding settings
+      - path: (.+)\.go$
+        text: 'field _struct is unused'
+        linters:
+          - unused
+        # ignore govet false positive fixed in https://github.com/golang/go/issues/45043
+      - path: (.+)\.go$
+        text: 'sigchanyzer: misuse of unbuffered os.Signal channel as argument to signal.Notify'
+        # ignore golint false positive fixed in https://github.com/golang/lint/pull/487
+      - path: (.+)\.go$
+        text: exported method (.*).Unwrap` should have comment or be unexported
```

Comment:
```suggestion
```
These should not be necessary, they were fixed a while ago and made their way into the linters